meetings/2021-07/july-14.md

```diff
@@ -0,0 +1,1028 @@
+
+AKI: It is worth noting to the procedural problem that since stage one is all about, you know, defining a problem as opposed to really carefully defining the solution to that problem it is debatable whether or not there's a procedural issue at all because this is potentially a solution to the problem that was brought.
+
+JRL: I'm okay with just asking for stage 1 for groupBy.
+
+MM: Thank you. I would feel uncomfortable -I understand the argument Aki made, but I would feel uncomfortable with that. So, I'm glad you're happy to go to stage one with groupBy.
+
+JRL: okay, so I think I'm with the resolution here. I'm definitely going to split groupBy into its own proposal, and then, I'm just asking stage 1 for `groupBy`.
+
+WH: Sounds good. I support stage 1 for `groupBy`. 
+
+MF: Great. I don't want to be difficult here, but like Aki was saying earlier, stage one is about defining a problem. Stage two is about selecting a solution. "groupBy for stage 1", I don't think is a meaningful statement here. We can be trying to solve just your filtering problem or trying to solve the problems that group by solves. We're not trying to group by if that makes sense.
+
+MM: Okay, groupBy solves a much bigger range of problems. So I would certainly. So maybe there's some writing that needs to happen before we can do this. But I would be willing to say that the problems that groupBy solves are well enough understood that I'm willing to say let's go to stage 1 on that set of problems with groupBy being the example approach for addressing those problems.
+
+MF:  I'm fine with that. Please Justin in your description in your repository, address the problem and not just the solution.
+
+WH: I take a different procedural position and I would say that, in my opinion, the `groupBy` proposal is almost at stage 2. We already have spec text for it, with the only modulo being that I would want the prototype gone from the produced objects and possibly a Map version. 
+
+JRL: So, I think with agreement that the Prototype object has to be changed to use object.create(null), or we have to use a Map, or a solution that does both. I think that's an agreement. There's also holes, which I think we still need to discuss but we can do that at stage one, and naming, which I think we're in agreement that just normal "groupBy" is fine. I think that is what I'm proposing to go to the stage one, that agreement
+
+JHD: I completely agree with the point making but I also think if we're going to consider both guilt or reject and groupBy separately, then they need to have two separate repositories for like Clarity and organization and whe named those proposals and what stage we decide there is a separate thing. But like, I think it's useful to have a specific repo that talks about group. I The problems at solving for the dresses, the Prototype thing and so on. And and I. So I think because of that, we need, it's prudent for us to separate it. Whether that's goes at stage one or two.
```

Comment:
```suggestion
JHD: I completely agree with the point making but I also think if we're going to consider both filterReject and groupBy separately, then they need to have two separate repositories for clarity and organization, and what we name those proposals and what stage we decide they are is a separate thing. But I think it's useful to have a specific repo that talks about groupBy, the problems it is solving or addresses, the prototype thing, and so on. So I think because of that, we need, it's prudent for us to separate it. Whether that goes at stage 1 or 2.
```